enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/context.tsx

```diff
@@ -0,0 +1,171 @@
+import {
+  createContext,
+  type PropsWithChildren,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { useUnmount } from "react-use";
+
+import type { SdkPluginsConfig } from "embedding-sdk";
+import { getDefaultVizHeight } from "embedding-sdk/lib/default-height";
+import { useSdkSelector } from "embedding-sdk/store";
+import { getPlugins } from "embedding-sdk/store/selectors";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { initializeQBRaw, resetQB } from "metabase/query_builder/actions";
+import {
+  getCard,
+  getFirstQueryResult,
+  getQueryResults,
+  getQuestion,
+  getUiControls,
+} from "metabase/query_builder/selectors";
+import type { Mode } from "metabase/visualizations/click-actions/Mode";
+import { getEmbeddingMode } from "metabase/visualizations/click-actions/lib/modes";
+import type Question from "metabase-lib/v1/Question";
+import type { Card, Dataset } from "metabase-types/api";
+import type { QueryBuilderUIControls } from "metabase-types/store";
+
+type InteractiveQuestionContextType = {
+  question: Question | undefined;
+  card: Card | null;
+  result: Dataset | null;
+  uiControls: QueryBuilderUIControls;
+  queryResults: Dataset[] | null;
+  plugins: SdkPluginsConfig | null;
+  mode: Mode | null | undefined;
+  defaultHeight?: number;
+  isQuestionLoading: boolean;
+  isQueryRunning: boolean;
+  resetQuestion: () => void;
+  onReset?: () => void;
+  onNavigateBack?: () => void;
+  withTitle?: boolean;
+  customTitle?: React.ReactNode;
+  withResetButton?: boolean;
+};
+
+export const InteractiveQuestionContext = createContext<
+  InteractiveQuestionContextType | undefined
+>(undefined);
+
+const returnNull = () => null;
+
+export const InteractiveQuestionProvider = ({
+  children,
+  location,
+  params,
+  componentPlugins,
+  onReset,
+  onNavigateBack,
+  withTitle = false,
+  customTitle,
+  withResetButton,
+}: PropsWithChildren<{
```

Comment:
Can you extract this into a type on its own like what other components do? You made the same mistake I did here, and it makes the component harder to read since the props are so long.
https://github.com/metabase/metabase/blob/bd2844c4a7e23818c409e7dd6ada4a0dd9056066/frontend/src/metabase/public/containers/PublicOrEmbeddedQuestion/PublicOrEmbeddedQuestionView.tsx#L40-L57

Reply:
done